### ISSUE TITLE
[static] redirect to cachable URL

### ISSUE THREAD

**Issue by trieloff** [adobe/helix-cli#95]
Static resources should have URLs that are unique per deployment. We deliver all static resources with long-lived `Cache-Control` headers, so we need to make sure that requests that point to the generic URL will get redirected to a stable URL that includes for instance the `ref`

**Comment by tripodsan** [comment]
the static resources processed by parcel will/would rename the static files based on their content sha. but for others we might do this in hlx.

**Comment by trieloff** [comment]
I'm happy to use Parcel's names (which are actually a bit better than changing the directory name, because they enable more granular caching). The tricky part will be injecting it back into the HTML in a way that does not force the developer to to strange stuff.

Maybe we could just rewrite links in the HTML that point to `/dist` or `/static`

**Comment by tripodsan** [comment]
as I said, parcel does this automatically for HTML. so far I wasn't able to trick parcel into process HTL as HTML and then process HTL into the JS template. I want to meet @devongovett next week to pick his brain, if possible.

**Comment by trieloff** [comment]
Maybe there is another path where Parcel processes our generated Javascript and detects (and re-writes) the dependencies there.

**Comment by tripodsan** [comment]
no, this would be much more difficult, to parse the `... out('<link src="/static/style.less">')...`

**Comment by trieloff** [comment]
Then I wonder if we can find a language-agnostic solution. We could provide a `staticroot` property as part of `payload.resource` so that you can inject it into your template or we could rewrite the generated HTML dynamically.
